XIVComboExpanded/Combos/NIN.cs

```diff
@@ -320,3 +320,60 @@ protected override uint Invoke(uint actionID, uint lastComboActionID, float comb
         return actionID;
     }
 }
+
+internal class KassatsuTrickDream : CustomCombo
+{
+    protected internal override CustomComboPreset Preset { get; } = CustomComboPreset.NinjaDreamyTrickAttack;
+    protected override uint Invoke(uint actionID, uint lastComboActionID, float comboTime, byte level)
+    {
+        if (actionID == NIN.Kassatsu)
+        {
+            if (!IsCooldownUsable(NIN.Kassatsu) && HasEffect(NIN.Buffs.ShadowWalker))
+                return OriginalHook(NIN.TrickAttack);
+        }
+
+        if (actionID == NIN.Kassatsu && !IsCooldownUsable(NIN.Kassatsu) && !IsCooldownUsable(NIN.TrickAttack))
+            return NIN.DreamWithinADream;
```

Comment:
You're duplicating a lot of logic in these conditionals.  Wouldn't this make more sense as:

```suggestion
        if (actionID == NIN.Kassatsu)
        {
            if (!IsCooldownUsable(NIN.Kassatsu)
            {
                if (HasEffect(NIN.Buffs.ShadowWalker))
                    return OriginalHook(NIN.TrickAttack);

                if (!IsCooldownUsable(NIN.TrickAttack))
                    return NIN.DreamWithinADream;
            }    
        }
```

Or even:

```suggestion
        if (actionID == NIN.Kassatsu && !IsCooldownUsable(NIN.Kassatsu)
        {
            if (HasEffect(NIN.Buffs.ShadowWalker))
                return OriginalHook(NIN.TrickAttack);

            if (!IsCooldownUsable(NIN.TrickAttack))
                return NIN.DreamWithinADream;
        }
```

I'd also recommend conditioning DreamWithinADream on it actually being off cooldown, though.  As written, you won't ever be able to see the remaining cooldown on Kassatsu unless Trick Attack is available but you lack Shadow Walker.  If you do the sequence as planned, it will show the remaining CD on Dream Within a Dream instead until Kassatsu actually comes off cooldown.